hack/crd-groups.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+# crd-groups generates crd or verify crd in CI
+
+if [ "$#" -lt 1 ] || [ "${1}" == "--help" ]; then
+  cat <<EOF
+Usage: $(basename $0) <action>
+
+  <action>        the action tu run ( generate, verify )
+
+Examples:
+  $(basename $0) generate
+EOF
+  exit 0
+fi
+
+ACTION="$1"
+shift 1
+
+GO_PKG="github.com/pingcap/tidb-operator"
+scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+to_crdgen="$scriptdir/../cmd/to-crdgen"
+crd_target="$scriptdir/../manifests/crd.yaml"
+crd_verify_target="$scriptdir/../manifests/crd-verify.yaml"
+
+
+GO111MODULE=on go get k8s.io/code-generator/cmd/openapi-gen@kubernetes-1.12.5
+
+$GOPATH/bin/openapi-gen --go-header-file=$scriptdir/boilerplate.go.txt \
+-i $GO_PKG/pkg/apis/pingcap.com/v1alpha1,k8s.io/apimachinery/pkg/apis/meta/v1,k8s.io/api/core/v1 \
+-p apis/pingcap.com/v1alpha1  -O openapi_generated -o $scriptdir/../pkg
+
+go install $to_crdgen
```

Comment:
add `GO111MODULE=off` prefix to avoid modifying `go.mod`, then you don't need to run `go mod tidy` at the end of the script

Reply:
my mistake, It seems that it's `go get k8s.io/code-generator/cmd/openapi-gen@kubernetes-1.12.5` that will modify go module files.

 I think we should vendor it in `go.mod`, and do not change `go.mod` in the script.

Reply:
I see the problem. The vendored version of code-generator is not 1.12.x because it has no go files under the root of the repository. So it cannot be vendored by `go mod` via `tools/tools.go`. We can fix these issues in the future.

Reply:
Temporarily fixed it by disable GO111MODULE for `go get k8s.io/code-generator/cmd/openapi-gen@kubernetes-1.12.5`